types/d3-array/d3-array-tests.ts

```diff
@@ -380,11 +567,17 @@ numbersArray = d3Array.range(1, 10, 0.5);
 // shuffle() -------------------------------------------------------------------
 
 mergedArray = d3Array.shuffle(mergedArray);
-
 mergedArray = d3Array.shuffle(mergedArray, 1);
-
 mergedArray = d3Array.shuffle(mergedArray, 1, 3);
 
+mergedArray = d3Array.shuffle(readonlyMergedArray);
```

Comment:
I suppose, the cleanest way here is to overload the `shuffle` signature with one signature for the traditional `array` as it originally was, and then adding one overload per type of Typed Array, e. g.:

```ts
/**
 * Randomizes the order of the specified array using the Fisher–Yates shuffle.
 */
export function shuffle(array: Float32Array, lo?: number, hi?: number): Float32Array;
```